enhancements/declarative-index-config.md

```diff
@@ -915,29 +1216,39 @@ rpc GetDefaultBundleThatProvides(GetDefaultProviderRequest) returns (Bundle) {}
 rpc ListBundles(ListBundlesRequest) returns (stream Bundle) {}
 ```
 
-Once the declarative package configs are available inside an index, these configs will be used to serve the content for these api endpoints instead of the sqllite database. A new flag `configs` will be introduced under `opm registry serve` that will accept a directory of config files as input, and use that directory to serve the grpc endpoints (instead of `registry serve --database index.db`).
+Once the declarative package configs are available inside an index, these configs will be used to serve the content for these api endpoints instead of the sqlite database.
+The `opm registry serve` will auto-detect the filetype of the `--database` flag and transparently handle both sqlite database files and declarative configs. If the database is formatted as sqlite, `opm registry serve` will internally migrate it to its declarative config representation prior to serving the GRPC API. When `opm registry serve` exits, the sqlite database will be preserved, and any temporary declarative config files will be cleaned up.
+
+## Deprecations
+
+Support for databases in the sqlite format will be marked for deprecation. All `opm` commands that interact with the database will print a deprecation warning on the
+command line when they detect an sqlite database is present. Users should plan to explicitly migrate their index images to the declarative config format because support for sqlite will be removed in a future release.
+
+Additionally, `prune`, `prune-stranded` and `rm` sub-commands under the `opm index` command will also be marked for deprecation.
```

Comment:
Not sure if we've decided on wholesale command deprecations yet, but this is present in the current merged enhancement, so leaving it in for now.

If possible and if this is the only sticking point, I propose we merge this and focus on deprecations in a follow-up.